addons/binding/org.openhab.binding.neato/src/main/java/org/openhab/binding/neato/handler/NeatoHandler.java

```diff
@@ -0,0 +1,255 @@
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        if (channelUID.getId().equals(org.openhab.binding.neato.NeatoBindingConstants.COMMAND)) {
+            logger.debug("Ok - will handle command for CHANNEL_COMMAND");
+            // TODO: handle command
+
+            try {
+                mrRobot.sendCommand(command);
+            } catch (Exception exc) {
+
+            }
+            // Note: if communication with thing fails for some reason,
+            // indicate that by setting the status with detail information
+            // updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+            // "Could not control device at IP address x.x.x.x");
+
+        }
+        this.refreshStateAndUpdate();
+    }
+
+    @Override
+    public void initialize() {
+
+        logger.info("Will boot up Neato Vacuum Cleaner binding!");
+        // TODO: Initialize the thing. If done set status to ONLINE to indicate proper working.
+
+        Configuration config = getThing().getConfiguration();
+        vacuumSecret = (String) config.get(CONFIG_SECRET);
+        vacuumName = (String) config.get(CONFIG_NAME);
+        vacuumSerialNumber = (String) config.get(CONFIG_SERIAL);
+
+        logger.info("Got config settings for {} with serial number: {}", vacuumName, vacuumSecret);
+
+        super.initialize();
+
+        try {
+            refreshTime = (BigDecimal) config.get(CONFIG_REFRESHTIME);
+            if (refreshTime.intValue() <= 0) {
+                throw new IllegalArgumentException("Refresh time must be positive number!");
+            }
+        } catch (Exception e) {
+            logger.warn("Refresh time [{}] is not valid. Falling back to default value: {}. {}", refreshTime,
+                    DEFAULTREFRESHTIME, e.getMessage());
+            refreshTime = DEFAULTREFRESHTIME;
+        }
+
+        mrRobot = new NeatoRobot(vacuumSerialNumber, vacuumSecret, vacuumName);
+        try {
+            if (mrRobot.sendGetState()) {
+                // Long running initialization should be done asynchronously in background.
+                updateStatus(ThingStatus.ONLINE);
+
+                // Get additional information - should we care about this?
+                mrRobot.sendGetRobotInfo();
+            }
+        } catch (Exception ex) {
+            // Note: When initialization can NOT be done set the status with more details for further
+            // analysis. See also class ThingStatusDetail for all available status details.
+            // Add a description to give user information to understand why thing does not work
+            // as expected. E.g.
+            // updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+            // "Can not access device as username and/or password are invalid");
+
+            logger.error(ex.getMessage());
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, ex.getMessage());
```

Comment:
If communication doesn't work to the server in initialisation time, thing is set to offline state. Does it auto recovery some how from this situation? If you move sendGetRotbotInfo to happen in background you could try to repeat it as long it success.